cmd/kueuectl/app/create/create_localqueue.go

```diff
@@ -0,0 +1,177 @@
+)
+
+const (
+	lqLong    = `Create a local queue with the given name in the specified namespace.`
+	lqExample = `  # Create a local queue 
+  kueuectl create localqueue my-local-queue -c my-cluster-queue`
+)
+
+type LocalQueueOptions struct {
+	PrintFlags *genericclioptions.PrintFlags
+
+	DryRunStrategy   util.DryRunStrategy
+	Name             string
+	Namespace        string
+	EnforceNamespace bool
+	ClusterQueue     v1beta1.ClusterQueueReference
+
+	UserSpecifiedClusterQueue string
+
+	Client kueuev1beta1.KueueV1beta1Interface
+
+	PrintObj printers.ResourcePrinterFunc
+
+	genericiooptions.IOStreams
+}
+
+func NewLocalQueueOptions(streams genericiooptions.IOStreams) *LocalQueueOptions {
+	return &LocalQueueOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("created").WithTypeSetter(scheme.Scheme),
+		IOStreams:  streams,
+	}
+}
+
+func NewLocalQueueCmd(clientGetter genericclioptions.RESTClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewLocalQueueOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:     "localqueue NAME --clusterqueue=cluster-queue-name [--dry-run=server|client|none]",
+		Aliases: []string{"lq"},
+		Short:   "Creates a localqueue",
+		Long:    lqLong,
+		Example: lqExample,
+		Args:    cobra.MatchAll(cobra.ExactArgs(1), cobra.OnlyValidArgs),
+		Run: func(cmd *cobra.Command, args []string) {
+			cobra.CheckErr(o.Complete(clientGetter, cmd, args))
+			cobra.CheckErr(o.Validate())
+			cobra.CheckErr(o.Run(cmd.Context()))
+		},
+	}
+
+	o.PrintFlags.AddFlags(cmd)
+
+	util.AddDryRunFlag(cmd)
+	cmd.Flags().StringVarP(&o.UserSpecifiedClusterQueue, "clusterqueue", "c", "",
+		"The cluster queue name which will be associated with the local queue (required).")
+	_ = cmd.MarkFlagRequired("clusterqueue")
+
+	return cmd
+}
+
+// Complete completes all the required options
+func (o *LocalQueueOptions) Complete(clientGetter genericclioptions.RESTClientGetter, cmd *cobra.Command, args []string) error {
+	o.Name = args[0]
+
+	var err error
+	o.Namespace, o.EnforceNamespace, err = clientGetter.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.ClusterQueue = v1beta1.ClusterQueueReference(o.UserSpecifiedClusterQueue)
+
+	config, err := clientGetter.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	clientset, err := versioned.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	o.Client = clientset.KueueV1beta1()
+
+	o.DryRunStrategy, err = util.GetDryRunStrategy(cmd)
+	if err != nil {
+		return err
+	}
+
+	err = util.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.DryRunStrategy)
+	if err != nil {
+		return err
+	}
+
+	printer, err := o.PrintFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+
+	o.PrintObj = printer.PrintObj
+
+	return nil
+}
+
+// Validate validates required fields are set to support structured generation
+func (o *LocalQueueOptions) Validate() error {
+	if len(o.Name) == 0 {
+		return fmt.Errorf("name must be specified")
+	}
+	if len(o.ClusterQueue) == 0 {
```

Comment:
It would be great if we also checked if the given clusterqueue actually exists.

Reply:
As I understood we are talking about --ignore-unknown-cq flag?

Reply:
Let's leave it for the next PR

Reply:
Made a new issue for it https://github.com/kubernetes-sigs/kueue/issues/2112.